src/fsharp/vs/service.fs

```diff
@@ -941,87 +1113,90 @@ type TypeCheckInfo
                     // note: as above, this happens when we are called for "precise" resolution - (F1 keyword, data tip etc..)
                     let plid, residue = List.frontAndBack origLongIdent
                     plid, Some residue
-                
+
+            let envItems, ty = GetEnvironmentLookupResolutions(mkPos line loc, plid, filterCtors, residueOpt.IsSome)
+            
             match nameResItems with            
             | NameResResult.TypecheckStaleAndTextChanged -> None // second-chance intellisense will try again
             | NameResResult.Cancel(denv,m) -> Some([], denv, m)
-            | NameResResult.Members(FilterRelevantItems exactMatchResidueOpt items) -> 
+            | NameResResult.Members(FilterRelevantItems exactMatchResidueOpt (items, denv, m)) -> 
                 // lookup based on name resolution results successful
-                Some items
+                Some (items |> List.map (CompletionItem ty), denv, m)
             | _ ->
-        
```

Comment:
Please remove the additional indentation of the code here, which I think is unrelated to the PR

It's really, really important try to minimize the diff as much as possible please.  This will need to be integrated into FSharp.COmpiler.Service.dll which still has some significant changes in service.fs.